lib/auth/auth.go

```diff
@@ -6078,10 +6078,25 @@ func (a *Server) isMFARequired(ctx context.Context, checker services.AccessCheck
 			services.NewWindowsLoginMatcher(t.WindowsDesktop.GetLogin()))
 
 	case *proto.IsMFARequiredRequest_App:
-		app, err := a.GetApp(ctx, t.App.GetName())
+		notFoundErr = trace.NotFound("application service %q not found", t.App.Name)
```

Comment:
Do we need to do this here? Grabbing a stack trace is not free, so why pay the cost of doing it if we don't know whether we will need the error?